### PR TITLE
Fix bad error message for FOR UPDATE on non-existent table

### PR DESCRIPTION
**Pull request by BaiShaoqi**
Fix github issue: https://github.com/greenplum-db/gpdb/issues/5915

### CODE REVIEW

**Comment by paul-guo-** [comment]
Looks good. Just one thing, you might want "github issue" instead of ambiguous word "issue" if you want the git issue number in the git log.

File: src/backend/parser/parse_relation.c [hunk 1]
```diff
@@ -1022,17 +1022,12 @@ addRangeTableEntry(ParseState *pstate,
...
 			
-			relid = RangeVarGetRelid(relation, lockmode, true);
-			if (relid == InvalidOid)
-				elog(ERROR, "Got Invalid RelationId of %s", relation->relname);
+			relid = RangeVarGetRelid(relation, lockmode, false);
 			
 			rel = try_heap_open(relid, NoLock, true);
 			if (!rel)
 				elog(ERROR, "open relation(%u) fail", relid);
-			if (locking->strength >= LCS_FORNOKEYUPDATE)
```
Comment (hlinnaka) [inline src/backend/parser/parse_relation.c]: This change to lockmode seems unrelated to the bad error message.

Comment (BaiShaoqi) [reply src/backend/parser/parse_relation.c]: Yes, the line 1032 is the same with 1021, so i think it do not need  to have two same  if clause.

Comment (hlinnaka) [reply src/backend/parser/parse_relation.c]: Ah, gotcha. Could simplify this further, and remove the `lockmode = (locking->strength >= LCS_FORNOKEYUPDATE) ? RowExclusiveLock : RowShareLock;` on line 1020, and add `else lockmode = RowShareLock` to after this block, at line 1037 (using the old line numbers, before this patch).

File: src/backend/parser/parse_relation.c [hunk 2]
```diff
@@ -1022,17 +1022,12 @@ addRangeTableEntry(ParseState *pstate,
...
-				lockmode = IsSystemRelation(rel) ? RowExclusiveLock : ExclusiveLock;
-			else
-				lockmode = RowShareLock;
```
Comment (hlinnaka) [inline src/backend/parser/parse_relation.c]: This `else` branch looks like it's actually misplaced, it probably should've been after the `}` below. We probably messed this up in one of the merges..
